tests/unit/lib/build_module/test_app_builder.py

```diff
@@ -325,6 +325,23 @@ def test_must_raise_on_docker_not_running(self):
         self.assertEquals(str(ctx.exception),
                           "Docker is unreachable. Docker needs to be running to build inside a container.")
 
+    @patch("samcli.lib.build.app_builder.supports_build_in_container")
+    def test_must_raise_on_unsupported_container_build(self, supports_build_in_container_mock):
+        config = Mock()
+
+        reason = "my reason"
+        supports_build_in_container_mock.return_value = (False, reason)
+
+        with self.assertRaises(ContainerBuildNotSupported) as ctx:
+            self.builder._build_function_on_container(config,
+                                                      "source_dir",
```

Comment:
do we need to pass in mode? we may need changes in `app_builder.py` as well.

Reply:
yeah, that's happening as part of the other PR that Jacob created